rotkehlchen/tests/exchanges/test_woo.py

```diff
@@ -0,0 +1,268 @@
+import warnings as test_warnings
+from json.decoder import JSONDecodeError
+from unittest.mock import call, patch
+
+import pytest
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.converters import asset_from_woo
+from rotkehlchen.constants.assets import A_BTC, A_ETH, A_SOL, A_USDT, A_WOO
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.exchanges.data_structures import (
+    AssetMovement,
+    AssetMovementCategory,
+    Trade,
+    TradeType,
+)
+from rotkehlchen.exchanges.woo import API_MAX_LIMIT, Woo
+from rotkehlchen.fval import FVal
+from rotkehlchen.types import Location, Timestamp
+from rotkehlchen.utils.misc import ts_sec_to_ms
+
+
+def test_name():
+    exchange = Woo('woo', 'a', b'a', object(), object())
+    assert exchange.location == Location.WOO
+    assert exchange.name == 'woo'
+
+
+@pytest.mark.parametrize(('start_ts', 'end_ts'), [(0, 1), (1634600000, 1634620000)])
+def test_query_online_trade_history_basic(mock_woo, start_ts, end_ts):
+    """Assert that the expected arguments are passed to the `_api_query` method"""
+    expected_call = call(
+        endpoint='v1/client/hist_trades',
+        options={
+            'end_t': ts_sec_to_ms(end_ts),
+            'fromId': 1,
+            'limit': API_MAX_LIMIT,
+            'start_t': ts_sec_to_ms(start_ts),
+        },
+    )
+    with patch.object(mock_woo, '_api_query') as mock_query:
+        mock_woo.query_online_trade_history(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+        )
+    assert mock_query.call_args_list == [expected_call]
+
+
+def test_query_balances(mock_woo):
```

Comment:
you also need:

1. A test of pagination for trades
2. A test of pagination for asset movements